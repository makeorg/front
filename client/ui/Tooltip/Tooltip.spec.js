import React from 'react';
import renderer from 'react-test-renderer';
import { Tooltip } from './index';

jest.mock('Client/ui/Elements/ButtonElements', () => ({
  TooltipWrapperStyle: 'TooltipWrapperStyle',
  TooltipStyle: 'TooltipStyle',
}));

jest.mock('Client/ui/Elements/ButtonElements', () => ({
  UnstyledButtonStyle: 'UnstyledButtonStyle',
}));

describe('TooltipWithTrigger', () => {
  const defaultProps = {
    triggerContent: 'foo',
    children: 'bar',
  };

  it('must match the snapshot with default Props', () => {
    const component = renderer.create(<Tooltip {...defaultProps} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
