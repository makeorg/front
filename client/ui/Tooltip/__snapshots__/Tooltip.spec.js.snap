// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipWithTrigger must match the snapshot with default Props 1`] = `
Array [
  <div
    aria-label="common.display_tooltip"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    bar
  </div>,
  .c0 {
  position: absolute;
  padding: 0.3125rem 0.625rem;
  background-color: rgb(51,51,51);
  color: rgb(255,255,255);
  font-size: 0.75rem;
  text-align: center;
}

.c0 >:after {
  content: '';
  position: absolute;
}

.c0[aria-hidden='true'] {
  display: none;
}

.c1 {
  position: absolute;
  padding: 0.3125rem 0.625rem;
  background-color: rgb(51,51,51);
  color: rgb(255,255,255);
  font-size: 0.75rem;
  text-align: center;
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
  left: 50%;
  bottom: calc(100% + 0.625rem);
}

.c1 >:after {
  content: '';
  position: absolute;
}

.c1[aria-hidden='true'] {
  display: none;
}

.c1 >:after {
  right: 50%;
  border-left: 0.3125rem solid transparent;
  border-right: 0.3125rem solid transparent;
  -webkit-transform: translate(50%,0);
  -ms-transform: translate(50%,0);
  transform: translate(50%,0);
}

.c1 >:after {
  top: 100%;
  border-top: 0.3125rem solid rgb(51,51,51);
}

<div
    aria-hidden={true}
    className="sc-12iuwgm-0 c0 c1"
    role="tooltip"
  />,
]
`;
